v-next/hardhat-keystore/test/helpers/mock-file-manager.ts

```diff
@@ -0,0 +1,75 @@
+import type {
+  FileManager,
+  UnencryptedKeystoreFile,
+} from "../../src/internal/types.js";
+import type { Mock } from "node:test";
+
+import { mock } from "node:test";
+
+import { assertHardhatInvariant } from "@ignored/hardhat-vnext-errors";
+
+import { UnencryptedKeystore } from "../../src/internal/keystores/unencrypted-keystore.js";
+
+export class MockFileManager implements FileManager {
+  #keystoreFile: UnencryptedKeystoreFile | null;
+
+  constructor() {
+    this.#keystoreFile = null;
+  }
+
+  /**
+   * Only used in tests to ensure there is no current keystore
+   * file at the path.
+   */
+  public setupNoKeystoreFile(): void {
+    this.#keystoreFile = null;
+  }
+
+  public setupExistingKeystoreFile(keys: {
+    [key: string]: string;
+  }): UnencryptedKeystoreFile {
+    const keystoreFile =
+      UnencryptedKeystore.createEmptyUnencryptedKeystoreFile();
+
+    keystoreFile.keys = keys;
+
+    this.#keystoreFile = keystoreFile;
+    this.writeJsonFile.mock.resetCalls();
+
+    return keystoreFile;
+  }
+
+  public setKeystoreFile(keystoreFile: UnencryptedKeystoreFile): void {
+    this.#keystoreFile = keystoreFile;
+  }
+
+  public fileExists: Mock<(_absolutePath: string) => Promise<boolean>> =
+    mock.fn(async (_absolutePath: string): Promise<boolean> => {
+      return this.#keystoreFile !== null;
+    });
+
+  public writeJsonFile: Mock<
```

Comment:
Some of the tests use `mock` from `node:test` to assert that a call wasn't made (i.e. write to file didn't happen).
Do we prefer an explicit custom `wasWriteJsonCalled` boolean to be added and updated, or are we happy with this `mock.fn` pattern?

Reply:
I think this is fine